docs/plugins/hook.rst

```diff
@@ -0,0 +1,77 @@
+Hook Plugin
+===============
+
+Internally, beets sends events to plugins when an action finishes. These can
+range from importing a song (``import``) to beets exiting (``cli_exit``), and
+provide a very flexible way to perform actions based on the events. This plugin
+allows you to run commands when an event is emitted by beets, such as syncing
+your library with another drive when the library is updated.
+
+Hooks are currently run in the order defined in the configuration, however this
+is dependent on beets itself (may change in the future) and cannot be controlled
+by this plugin.
+
+.. _hook-configuration:
+
+Configuration
+-------------
+
+To configure the plugin, make a ``hook:`` section in your configuration
+file. The available options are:
+
+- **hooks**: A list of events and the commands to run (see :ref:`individual-hook-configuration`)
+  Default: Empty.
+- **substitute_event**: The string to replace in each command with the name of
+  the event executing it. This can be used to allow one script to act
+  differently depending on the event it was called by. Can be individually
+  overridden (see :ref:`individual-hook-configuration`).
+  Default: ``%EVENT%``
+- **shell**: Run each command in a shell. Can be individually overridden (see
+  :ref:`individual-hook-configuration`).
+  Default: ``yes``
```

Comment:
After doing some research, I'm not entirely sure. `shell` is platform dependant, and also depends on the user's `$SHELL`. After reading [this Stack Overflow answer](http://stackoverflow.com/questions/3172470/actual-meaning-of-shell-true-in-subprocess) I'm thinking we should make it `no` by default and let people turn it on if they need it. What do you think?